vars/healthExecuteCheck.groovy

```diff
@@ -0,0 +1,39 @@
+import com.sap.piper.ConfigurationHelper
+
+import groovy.transform.Field
+
+@Field String STEP_NAME = 'healthExecuteCheck'
+@Field Set STEP_CONFIG_KEYS = ['healthEndpoint', 'testServerUrl']
+@Field Set PARAMETER_KEYS = STEP_CONFIG_KEYS
+
+void call(Map parameters = [:]) {
+    handlePipelineStepErrors (stepName: STEP_NAME, stepParameters: parameters) {
+        def script = parameters?.script ?: [commonPipelineEnvironment: commonPipelineEnvironment]
+        // load default & individual configuration
+        Map config = ConfigurationHelper
+            .loadStepDefaults(this)
+            .mixinGeneralConfig(script.commonPipelineEnvironment, STEP_CONFIG_KEYS)
+            .mixinStepConfig(script.commonPipelineEnvironment, STEP_CONFIG_KEYS)
+            .mixinStageConfig(script.commonPipelineEnvironment, parameters.stageName?:env.STAGE_NAME, STEP_CONFIG_KEYS)
+            .mixin(parameters, PARAMETER_KEYS)
+            .withMandatoryProperty('testServerUrl')
+            .use()
+
+        def delimiter = config.testServerUrl.endsWith('/') ? '' : '/'
+        def checkUrl = config.testServerUrl + (config.healthEndpoint ? "${delimiter}${config.healthEndpoint}" : '')
+
+        def statusCode = curl(checkUrl)
+        if (statusCode != '200') {
```

Comment:
[nit] What about other return values in the 200-Range?

Reply:
they are not really relevant for the health of an application. A standard health endpoint should provide 200 if everything is okay. 
This endpoint will also potentially be used by load balancers, ... that don't have a lot of logic for health checking ...